fitbenchmarking/core/tests/test_results_output.py

```diff
@@ -1,47 +1,346 @@
+              [7, 3, 5, 1]]
+    min_runtime = [4.2e-5, 5.0e-14]
+    runtime_in = [[1e-2, 2.2e-3, 4.2e-5, 9.8e-1],
+                  [3.0e-10, 5.0e-14, 1e-7, 4.3e-12]]
+
+    results_out = []
+    for i, p in enumerate(problems):
+        p.data_x = data_x[i]
+        p.data_y = data_y[i]
+        p.data_e = data_e[i]
+        p.function = func[i]
+        p.name = "prob_{}".format(i)
+        results = []
+        for j in range(num_min):
+            p.starting_values = starting_values
+            jac = Scipy(p)
+            jac.method = '2-point'
+            r = FittingResult(options=options, problem=p, jac=jac,
+                              initial_params=starting_values,
+                              params=params_in[i][j])
+            r.chi_sq = acc_in[i][j]
+            r.runtime = runtime_in[i][j]
+            r.error_flag = error_in[i][j]
+            r.support_page_link = link_in[i][j]
+            r.minimizer = options.minimizers[software][j]
+            results.append(r)
+        results_out.append(results)
+    return results_out, options, min_chi_sq, min_runtime
 
 
 class SaveResultsTests(unittest.TestCase):
-    def test_dummy(self):
+    """
+    Unit tests for save_results function
+    """
+
+    def setUp(self):
         """
-        Dummy test to appease pytest
+        Setting up paths and results folders
+        """
+        self.results, self.options, self.min_chi_sq, self.min_runtime = \
+            generate_mock_results()
+        test_path = os.path.dirname(os.path.realpath(__file__))
+        self.dirname = os.path.join(test_path, 'results')
+        self.options.results_dir = self.dirname
+        os.mkdir(self.dirname)
+
+    def tearDown(self):
+        """
+        Clean up created folders.
         """
-        pass
+        shutil.rmtree(self.dirname)
+
+    def test_save_results_correct(self):
+        """
+        Tests to check the group_dir is correct
+        """
+        failed_problems = []
+        unselected_minimzers = {}
+        group_name = "group_name"
+        group_dir = save_results(self.options, self.results, group_name,
+                                 failed_problems, unselected_minimzers)
+        assert group_dir == os.path.join(self.dirname, group_name)
 
 
 class CreateDirectoriesTests(unittest.TestCase):
-    def test_dummy(self):
+    """
+    Unit tests for create_directories function
+    """
+
+    def setUp(self):
+        """
+        Setting up paths and results folders
+        """
+        self.options = Options()
+        test_path = os.path.dirname(os.path.realpath(__file__))
+        self.dirname = os.path.join(test_path, 'results')
+        os.mkdir(self.dirname)
+
+    def tearDown(self):
+        """
+        Clean up created folders.
+        """
+        shutil.rmtree(self.dirname)
+
+    def test_create_dirs_correct(self):
         """
         Dummy test to appease pytest
         """
-        pass
+        group_name = 'test_group'
```

Comment:
I think this docstring needs updating

Reply:
Good spot, I have corrected the docstring